integration-tests/src/common/mod.rs

```diff
@@ -118,16 +117,145 @@ pub(crate) fn fund_address(address: &Address<NetworkChecked>) -> eyre::Result<()
     Ok(())
 }
 
+/* Let's fork bitcoincore-rpc instead for BitVM branch compatability */
+
+pub(crate) fn create_wallet(
+    client: &Client,
+    name: &str,
+) -> Result<serde_json::Value, jsonrpc::Error> {
+    client.call("createwallet", Some(&to_raw_value(&[name]).unwrap()))
+}
+
+pub(crate) fn get_new_address(
+    client: &Client,
+    label: &str,
+) -> Result<Address<NetworkUnchecked>, jsonrpc::Error> {
+    client.call(
+        "getnewaddress",
+        Some(&to_raw_value(&[label, "bech32m"]).unwrap()),
+    )
+}
+
+pub(crate) fn get_balance(client: &Client) -> Result<Amount, jsonrpc::Error> {
+    let number: f64 = client.call("getbalance", None)?;
+
+    Ok(Amount::from_float_in(number, Denomination::Bitcoin).unwrap())
+}
+
+pub(crate) fn get_block_count(client: &Client) -> Result<usize, jsonrpc::Error> {
+    client.call("getblockcount", None)
+}
+
+pub(crate) fn generate_to_address(
+    client: &Client,
+    blocks: usize,
+    address: Address,
+) -> Result<serde_json::Value, jsonrpc::Error> {
+    client.call(
+        "generatetoaddress",
+        Some(
+            &to_raw_value(&[
+                to_raw_value(&blocks).unwrap(),
+                to_raw_value(&address).unwrap(),
+            ])
+            .unwrap(),
+        ),
+    )
+}
+
+pub(crate) fn fund_raw_transaction(
+    client: &Client,
+    tx: &Transaction,
+    change_pos: isize,
+) -> Result<Transaction, jsonrpc::Error> {
+    let hextx = {
+        // TODO(Velnbur): only this works and not this:
+        // `bitcoin::consensus::encode::serialize_hex(tx)` because
+        // fuck, bitcoin, I don't know...
+        let mut buff = Vec::new();
+        tx.version.consensus_encode(&mut buff).unwrap();
+        // 0x00.consensus_encode(&mut buff).unwrap();
```

Comment:
Are these comments needed?

Reply:
In theory, if somebody else will struggle with an encoding of txs in RPC calls. I can  remove it

Reply:
Just add a comment, "If someone is struggling with..., here is..." or alike